src/navigation.js

```diff
@@ -2,18 +2,26 @@ const { testStep } = require('../utils/supertest');
 const httpStatus = require('http-status-codes');
 
 const navigatesToNext = (step, nextStep, session) => {
-  const test = testStep(step)
+  const stepType = Object.getPrototypeOf(step).name;
+
+  let test = testStep(step)
     .withSteps(nextStep);
 
   if (session) {
     test.withSession(session);
-  } else {
+  } else if (stepType !== 'EntryPoint'){
     test.withSetup(req => {
       return req.session.generate();
     });
   }
 
-  return test.post()
+  if (stepType === 'Redirect' || stepType === 'EntryPoint') {
```

Comment:
Personally, I would prefer having specific functions with an appropriate name for kind of step but maybe this is more backward compatible

Reply:
i thought it better just to have one test for navigation that can handle each type of step, given this is the only difference...